File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    let width=0, height=0;
    
    for (let size of sizes) {
        const w = Math.max(...size);
        const h = Math.min(...size);
        
        width = Math.max(w, width);
        height = Math.max(h, height);
    }
    
    return width*height;
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    min_ = 0
    max_ = 0
    
    for w,h in sizes:
        if w>h:
            max_ = max(max_, w)
            min_ = max(min_, h)
        else:
            max_ = max(max_, h)
            min_ = max(min_, w)
            
    return max_*min_